### ISSUE TITLE
clx article fails on items with escaped backslashes: "invalid character 'n' in \u hexadecimal character escape"

### ISSUE THREAD

**Issue by ivoronin** [bensadeh/circumflex#201]
### Repro

https://news.ycombinator.com/item?id=48264635

>  All commands have the format `output = \func inputs` or just `\function inputs`. Points and scalars are built on the fly. Eg `\line a b` to an empty canvas creates points `a` and `b`, and joins them with a line.One can use broadcasting semantics similar to NumPy and PyTorch in a visual setting (imagine creating a list of circles where one dim corresponds to radius and another to the center). One can also use backpropagation, run gradient descent or visualize vector fields. Almost everything is reactive so changing a variable updates all of the downstream geometry. It also allows anyone to write and load their own visualization, which can be broadcasted and differentiated through.

```
$ clx article 48264635
Error: unexpected response from server for item 48264635: invalid character 'n' in \u hexadecimal character escape
```

### Cause

`ansi.Strip` is applied to the raw JSON response body before parsing (`hn/services/firebase/firebase.go:263`). The `dangerousControls` regex matches `\b`/`\f` JSON short escapes textually without understanding JSON escape semantics, so an escaped backslash followed by `b`/`f` is corrupted.

Item 48264635's `text` field contains the LaTeX-style command `\func`, which the API serializes as `\\func` (raw bytes `\ \ f u n c`). The regex matches `\f` starting at the second backslash, dropping it. `\\func` → `\unc`, and JSON parsing then fails on `\un` (`n` is not a hex digit).

**Comment by bensadeh** [comment]
Thanks for the detailed explanation and pinpointing the error exactly! Fixed in 6902770ebf9f53f417295c6fa526d54478045e8c.
